packages/discovery-react-components/src/components/SearchResults/SearchResults.tsx

```diff
@@ -73,6 +73,10 @@ export interface SearchResultsProps {
    * override default messages for the component by specifying custom and/or internationalized text strings
    */
   messages?: Partial<Messages>;
+  /**
+   * callback function from the component for sending document
+   */
+  handleSelectedResult?: (document: DiscoveryV2.QueryResult) => void | undefined;
```

Comment:
why do we need this? can't you utilize the `SearchContext` to check for a change to the `selectedResult`? adding this to the top level means we have to support it forever until we decide to make a breaking change, so would like to understand the use-case better.

also we should add a test for this

Reply:
So after creating a new route for rendering the document, we observed a lag in `searchContext` and getting the document from there. So instead of relying on `SearchContext` change wanted to add a specific callback to access the document.